### ISSUE TITLE
Add validation set and callbacks in pretrain and train scripts

### ISSUE THREAD

**Issue by hardik124** [marco-c/autowebcompat#150]
Currently, we only have train accuracy and its not being validated. We can go for a simple validation script or a K fold validation. 
With validation check , we can add callbacks such as early stopping and model checkpoints.

**Comment by marco-c** [comment]
I think simple validation is preferable. K-fold is not feasible, the training takes too much time.

**Comment by hardik124** [comment]
@marco-c Can i take this up ?

**Comment by marco-c** [comment]
Until there is a PR for an issue, the issue is available for anyone to work on it.
